### PR TITLE
added category model to project

### PR DESCRIPTION
**Pull request by yuriy-kononchuk**

### CODE REVIEW

File: src/main/java/com/example/project/repository/book/BookRepository.java [hunk 1]
```diff
@@ -1,9 +1,10 @@
 package com.example.project.repository.book;
 
 import com.example.project.model.Book;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
-
+    List<Book> findAllByCategoriesId(Long categoryId); // added
```
Comment (olekskov) [inline src/main/java/com/example/project/repository/book/BookRepository.java:9]: please remove such comments, they are unnecessary

```suggestion
    List<Book> findAllByCategoriesId(Long categoryId);
```

**Review by olekskov — APPROVED** [review]

File: src/main/java/com/example/project/repository/book/BookRepository.java (hunk 1 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/repository/book/BookRepository.java:9]: sure

File: src/main/java/com/example/project/repository/CartItemRepository.java [hunk 2]
```diff
@@ -6,5 +6,5 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
-    List<CartItem> findAllByUsersId(Long userId, Pageable pageable);
+    List<CartItem> findAllByShoppingCart_User_Id(Long userId, Pageable pageable);
```
Comment (okuzan) [inline src/main/java/com/example/project/repository/CartItemRepository.java]: we don't use _ in method names unless for tests, why did you change the name?

File: src/main/resources/db/changelog/changes/12-add-cartitem-table.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                  name: quantity
+                  type: bigint
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/12-add-cartitem-table.yaml]: image
add empty line everywhere you see this symbol

Comment (yuriy-kononchuk) [reply src/main/resources/db/changelog/changes/12-add-cartitem-table.yaml]: done

File: src/main/java/com/example/project/repository/CartItemRepository.java (hunk 2 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/repository/CartItemRepository.java]: we've discussed this issue. added changes in naming and methods using shoppingCartId approach

File: src/main/java/com/example/project/mapper/OrderItemMapper.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import com.example.project.config.MapperConfig;
+import com.example.project.dto.orderitem.CreateOrderItemRequestDto;
+import com.example.project.dto.orderitem.OrderItemDto;
+import com.example.project.model.OrderItem;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class, uses = {OrderMapper.class, BookMapper.class})
+public interface OrderItemMapper {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/mapper/OrderItemMapper.java:12]: delete empty line

File: src/main/java/com/example/project/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PatchMapping("/{id}")
+    @Operation(summary = "Update status of order", description = "Update status of order")
+    @ApiResponse(responseCode = "200", description = "Order's status was successfully updated")
+    public OrderDto updateStatusOrder(@PathVariable Long id, Order.Status status,
+                                      Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.updateStatus(user, id, status);
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/controller/OrderController.java:60]: delete empty line

File: src/main/java/com/example/project/mapper/OrderMapper.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.Optional;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.Named;
+
+@Mapper(config = MapperConfig.class)
+public interface OrderMapper {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/mapper/OrderMapper.java:17]: delete empty line

File: src/main/java/com/example/project/repository/OrderRepository.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.example.project.repository;
+
+import com.example.project.model.Order;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface OrderRepository extends JpaRepository<Order, Long> {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/repository/OrderRepository.java:9]: delete empty line

File: src/main/java/com/example/project/service/OrderItemService.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.example.project.service;
+
+import com.example.project.dto.orderitem.CreateOrderItemRequestDto;
+import com.example.project.dto.orderitem.OrderItemDto;
+import com.example.project.model.User;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
+
+public interface OrderItemService {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/service/OrderItemService.java:10]: delete empty line

File: src/main/java/com/example/project/service/OrderService.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.example.project.dto.order.CreateOrderRequestDto;
+import com.example.project.dto.order.OrderDto;
+import com.example.project.model.Order;
+import com.example.project.model.User;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
+
+public interface OrderService {
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/service/OrderService.java:11]: delete empty line

**Review by kozhukhovsky — APPROVED** [review]

File: src/main/java/com/example/project/mapper/OrderItemMapper.java (hunk 4 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/mapper/OrderItemMapper.java:12]: done

File: src/main/java/com/example/project/controller/OrderController.java (hunk 5 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/controller/OrderController.java:60]: done

File: src/main/java/com/example/project/mapper/OrderMapper.java (hunk 6 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/mapper/OrderMapper.java:17]: done

File: src/main/java/com/example/project/repository/OrderRepository.java (hunk 7 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/repository/OrderRepository.java:9]: done

File: src/main/java/com/example/project/service/OrderItemService.java (hunk 8 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/service/OrderItemService.java:10]: done

File: src/main/java/com/example/project/service/OrderService.java (hunk 9 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/service/OrderService.java:11]: done
